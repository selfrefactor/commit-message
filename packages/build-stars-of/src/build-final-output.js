import { flip, interpolate, join, map, piped } from 'rambdax'

function getIntro(title, repo){
  const template = `
# {{title}}

This list contains Github repos depending on **{{repo}}**. 

These repos are sorted by their stars and thier \`package.json\` is updated in the last year.

> This list is created with \`build-stars-of\` library.
  `.trim()

  return interpolate(template, {
    title,
    repo,
  })
}

function getMainContent(data){
  const template = `
## {{name}}

{{description}}

[{{name}}](https://github.com/{{name}}) - ðŸŒŸ {{stars}}
  `.trim()

  const templateNoDescription = `
## {{name}}

[{{name}}](https://github.com/{{name}}) - ðŸŒŸ {{stars}}
  `.trim()

  const allMethods = piped(
    data,
    map(({ repoData: x }) =>
      interpolate(x.description ? template: templateNoDescription, {
        name        : x.full_name,
        description : x.description,
        stars       : x.stargazers_count,
      })),
    join('\n\n')
  )

  return allMethods
}

export function buildFinalOutput(input){
  const template = `
{{intro}}

{{mainContent}}
  `.trim()
  const intro = getIntro(input.title, input.repo)

  const mainContent = getMainContent(input.data)

  const markdownText = interpolate(template, {intro, mainContent})

  return markdownText
}
