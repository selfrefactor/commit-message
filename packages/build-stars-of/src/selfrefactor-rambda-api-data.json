{
  "data": [
    {
      "repoData": {
        "full_name": "BeeeQueue/yuna",
        "description": "ãƒ„ An anime player that integrates with AniList, Simkl, Crunchyroll, and Hidive.",
        "stargazers_count": 313,
        "forks_count": 36,
        "open_issues_count": 50,
        "pushed_at": "2020-12-22T04:15:51Z",
        "updated_at": "2020-12-23T17:11:57Z",
        "subscribers_count": 14
      },
      "repoUrl": "BeeeQueue/yuna",
      "filterData": {
        "pass": true,
        "updateDate": "2020-12-21T12:40:35Z",
        "updateDiff": 6
      }
    },
    {
      "repoData": {
        "full_name": "caderek/benny",
        "description": "A dead simple benchmarking framework for JS/TS libs",
        "stargazers_count": 274,
        "forks_count": 4,
        "open_issues_count": 6,
        "pushed_at": "2020-12-22T09:26:08Z",
        "updated_at": "2020-12-13T17:47:44Z",
        "subscribers_count": 4
      },
      "repoUrl": "caderek/benny",
      "filterData": {
        "pass": true,
        "updateDate": "2020-11-13T21:43:27Z",
        "updateDiff": 44
      }
    },
    {
      "repoData": {
        "full_name": "farwayer/mobx-decorators",
        "description": "Several helper MobX decorators (setter, observe, save, ...)",
        "stargazers_count": 156,
        "forks_count": 4,
        "open_issues_count": 7,
        "pushed_at": "2020-12-11T03:30:09Z",
        "updated_at": "2020-11-04T15:43:31Z",
        "subscribers_count": 7
      },
      "repoUrl": "farwayer/mobx-decorators",
      "filterData": {
        "pass": false,
        "updateDate": "2019-11-05T16:41:01Z",
        "updateDiff": 418
      }
    },
    {
      "repoData": {
        "full_name": "selfrefactor/rambdax",
        "description": "Extended version of Rambda ",
        "stargazers_count": 136,
        "forks_count": 18,
        "open_issues_count": 1,
        "pushed_at": "2020-11-25T12:32:09Z",
        "updated_at": "2020-12-27T14:34:52Z",
        "subscribers_count": 4
      },
      "repoUrl": "selfrefactor/rambdax",
      "filterData": {
        "pass": true,
        "updateDate": "2020-11-25T12:31:55Z",
        "updateDiff": 32
      }
    },
    {
      "repoData": {
        "full_name": "subfuzion/voting-app",
        "description": "Example containerized microservices Voting App based on the original Docker version",
        "stargazers_count": 110,
        "forks_count": 48,
        "open_issues_count": 7,
        "pushed_at": "2018-11-10T08:24:27Z",
        "updated_at": "2020-11-05T22:35:30Z",
        "subscribers_count": 11
      },
      "repoUrl": "subfuzion/voting-app",
      "filterData": {
        "pass": false
      }
    },
    {
      "repoData": {
        "full_name": "romgrk/todoist.nvim",
        "description": "A todoist extension for neovim",
        "stargazers_count": 72,
        "forks_count": 2,
        "open_issues_count": 7,
        "pushed_at": "2020-11-26T18:22:10Z",
        "updated_at": "2020-12-21T02:05:25Z",
        "subscribers_count": 5
      },
      "repoUrl": "romgrk/todoist.nvim",
      "filterData": {
        "pass": true,
        "updateDate": "2020-06-06T18:22:40Z",
        "updateDiff": 204
      }
    },
    {
      "repoData": {
        "full_name": "SAP-samples/teched2020-developer-keynote",
        "description": "SAP TechEd 2020: Developer Keynote",
        "stargazers_count": 64,
        "forks_count": 54,
        "open_issues_count": 1,
        "pushed_at": "2020-12-24T14:33:27Z",
        "updated_at": "2020-12-26T16:26:47Z",
        "subscribers_count": 15
      },
      "repoUrl": "SAP-samples/teched2020-developer-keynote",
      "filterData": {
        "pass": false
      }
    }
  ]
}
