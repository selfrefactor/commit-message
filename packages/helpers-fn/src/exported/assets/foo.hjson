const FOO = a 1 b 2

{
  a: node {{FOO}}
}