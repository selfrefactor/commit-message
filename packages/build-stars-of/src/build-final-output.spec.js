import { data } from './assets/selfrefactor-rambda-api-data'
import { buildFinalOutput } from './build-final-output'

test('happy', () => {
  const input = {
    repo  : 'selfrefactor/rambda',
    title : 'Stars of **Rambda** list',
    data,
  }
  expect(buildFinalOutput(input)).toMatchSnapshot()
})
