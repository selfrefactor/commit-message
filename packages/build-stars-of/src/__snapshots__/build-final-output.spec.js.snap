// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`happy 1`] = `
"# Stars of **Rambda** list

This list contains Github repos depending on **selfrefactor/rambda**. 

These repos are sorted by their stars and thier \`package.json\` is updated in the last year.

> This list is created with \`build-stars-of\` library.

## BeeeQueue/yuna

ãƒ„ An anime player that integrates with AniList, Simkl, Crunchyroll, and Hidive.

[BeeeQueue/yuna](https://github.com/BeeeQueue/yuna) - ðŸŒŸ 313

## caderek/benny

A dead simple benchmarking framework for JS/TS libs

[caderek/benny](https://github.com/caderek/benny) - ðŸŒŸ 274

## farwayer/mobx-decorators

Several helper MobX decorators (setter, observe, save, ...)

[farwayer/mobx-decorators](https://github.com/farwayer/mobx-decorators) - ðŸŒŸ 156

## selfrefactor/rambdax

Extended version of Rambda 

[selfrefactor/rambdax](https://github.com/selfrefactor/rambdax) - ðŸŒŸ 136

## subfuzion/voting-app

Example containerized microservices Voting App based on the original Docker version

[subfuzion/voting-app](https://github.com/subfuzion/voting-app) - ðŸŒŸ 110

## selfrefactor/useful-javascript-libraries

Collection of Javascript libraries

[selfrefactor/useful-javascript-libraries](https://github.com/selfrefactor/useful-javascript-libraries) - ðŸŒŸ 80

## romgrk/todoist.nvim

A todoist extension for neovim

[romgrk/todoist.nvim](https://github.com/romgrk/todoist.nvim) - ðŸŒŸ 72

## SAP-samples/teched2020-developer-keynote

SAP TechEd 2020: Developer Keynote

[SAP-samples/teched2020-developer-keynote](https://github.com/SAP-samples/teched2020-developer-keynote) - ðŸŒŸ 64

## SAP/cloud-sdk-js

Use the SAP Cloud SDK for JavaScript / TypeScript to reduce development effort when building applications on SAP Cloud Platform that communicate with SAP solutions and services such as SAP S/4HANA Cloud, SAP SuccessFactors, and many others. 

[SAP/cloud-sdk-js](https://github.com/SAP/cloud-sdk-js) - ðŸŒŸ 58

## sapmentors/cds-pg

PostgreSQL adapter for SAP CDS (CAP)

[sapmentors/cds-pg](https://github.com/sapmentors/cds-pg) - ðŸŒŸ 55

## aws-samples/voteapp

Cloud native microservices app running with a service mesh (AWS App Mesh).

[aws-samples/voteapp](https://github.com/aws-samples/voteapp) - ðŸŒŸ 50

## nicobarray/aquedux

Redux over the wire

[nicobarray/aquedux](https://github.com/nicobarray/aquedux) - ðŸŒŸ 46

## HumanCellAtlas/metadata-schema

This repo is for the metadata schemas associated with the HCA

[HumanCellAtlas/metadata-schema](https://github.com/HumanCellAtlas/metadata-schema) - ðŸŒŸ 44

## vmarcosp/covidbr-app

COVIDBR App  â€”â€” Dados sobre os casos de corona vÃ­rus no Brasil

[vmarcosp/covidbr-app](https://github.com/vmarcosp/covidbr-app) - ðŸŒŸ 42

## SAP-samples/hana-developer-cli-tool-example

Learn how to build a developer-centric SAP HANA command line tool, particularly designed to be used when performing SAP HANA development in non-SAP tooling (for example from VS Code).

[SAP-samples/hana-developer-cli-tool-example](https://github.com/SAP-samples/hana-developer-cli-tool-example) - ðŸŒŸ 31

## SAP-samples/ui5-cap-event-app

Showcase of SAP Cloud Application Programming Model and OData V4 with draft mode in a freestyle SAPUI5 app and an SAP Fiori elements app.

[SAP-samples/ui5-cap-event-app](https://github.com/SAP-samples/ui5-cap-event-app) - ðŸŒŸ 29

## vmarcosp/covidbr-api

COVIDBR API â€”â€” API de tracking dos casos de COVID-19 no Brasil

[vmarcosp/covidbr-api](https://github.com/vmarcosp/covidbr-api) - ðŸŒŸ 29

## SAP/cloud-sdk-cli

The command line interface for the SAP Cloud SDK

[SAP/cloud-sdk-cli](https://github.com/SAP/cloud-sdk-cli) - ðŸŒŸ 26

## selfrefactor/services

Holder of multiple npm packages

[selfrefactor/services](https://github.com/selfrefactor/services) - ðŸŒŸ 25

## zyxd/svelte-store-router

Store-based router for Svelte

[zyxd/svelte-store-router](https://github.com/zyxd/svelte-store-router) - ðŸŒŸ 24

## k-yokoishi/zsh-prompt-generator

Zsh Prompt Generator

[k-yokoishi/zsh-prompt-generator](https://github.com/k-yokoishi/zsh-prompt-generator) - ðŸŒŸ 24

## jpgorman/react-append-to-body

React Higher order component that allows you to attach components to the DOM outside of the main app.

[jpgorman/react-append-to-body](https://github.com/jpgorman/react-append-to-body) - ðŸŒŸ 24

## roblav96/robinhood.tools

ðŸ“ˆðŸ¤‘ðŸ’° Advanced trading tools and resources for Robinhood Web.

[roblav96/robinhood.tools](https://github.com/roblav96/robinhood.tools) - ðŸŒŸ 21

## mischnic/tree-shaking-example

Tree-shaking of Rambda, Ramda and Lodash with Webpack, Parcel, esbuild and Rollup

[mischnic/tree-shaking-example](https://github.com/mischnic/tree-shaking-example) - ðŸŒŸ 20

## farwayer/mst-decorators

Class based MobX-State-Tree definitions

[farwayer/mst-decorators](https://github.com/farwayer/mst-decorators) - ðŸŒŸ 20

## verydanny/vcslack

VSCode Slack intergration

[verydanny/vcslack](https://github.com/verydanny/vcslack) - ðŸŒŸ 18

## devilcoders/tachyons-for-js

Tachyons for CSS-in-JS

[devilcoders/tachyons-for-js](https://github.com/devilcoders/tachyons-for-js) - ðŸŒŸ 18

## gregorwolf/pg-beershop

CAP Beershop with PostgreSQL

[gregorwolf/pg-beershop](https://github.com/gregorwolf/pg-beershop) - ðŸŒŸ 16

## LykkeCity/LykkeTerminalWeb

null

[LykkeCity/LykkeTerminalWeb](https://github.com/LykkeCity/LykkeTerminalWeb) - ðŸŒŸ 16

## SAP-samples/cloud-cap-multitenancy

SAP Cloud Application Programming Model (CAP) sample code project with multitenancy using service manager-created SAP HANA containers for tenant data isolation.

[SAP-samples/cloud-cap-multitenancy](https://github.com/SAP-samples/cloud-cap-multitenancy) - ðŸŒŸ 15

## solui/solui

Generate DApps for smart contracts

[solui/solui](https://github.com/solui/solui) - ðŸŒŸ 15

## wagerfield/ig-api

Axios wrapper around IG's API

[wagerfield/ig-api](https://github.com/wagerfield/ig-api) - ðŸŒŸ 15

## SAP-samples/cloud-messaging-handsonsapdev

For the host and participants of the Hands-on SAP Dev live stream show to share discoveries and scripting efforts in the Diving into messaging on SAP Cloud Platform series.

[SAP-samples/cloud-messaging-handsonsapdev](https://github.com/SAP-samples/cloud-messaging-handsonsapdev) - ðŸŒŸ 14

## jesster2k10/react-native-ts-boilerplate

A easy to use, fully featured, React Native Boilerplate that helps you save time.

[jesster2k10/react-native-ts-boilerplate](https://github.com/jesster2k10/react-native-ts-boilerplate) - ðŸŒŸ 14

## Ovyerus/license

Easily generate licenses for your projects!

[Ovyerus/license](https://github.com/Ovyerus/license) - ðŸŒŸ 14

## SAP-samples/cloud-foundry-cap-azure-cities

Reference application built according to the SAP Cloud Application Programming Model. It uses an SAP HANA database for structured data, an Azure storage account for unstructured data, and SAP Fiori Elements for the user interface.

[SAP-samples/cloud-foundry-cap-azure-cities](https://github.com/SAP-samples/cloud-foundry-cap-azure-cities) - ðŸŒŸ 14

## the-dr-lazy/rostam-bot-extension

An extension that notify you suspicious accounts on Twitter in Chrome and Firefox.

[the-dr-lazy/rostam-bot-extension](https://github.com/the-dr-lazy/rostam-bot-extension) - ðŸŒŸ 14

## SAP-samples/teched2020-DEV267

DEV267 - Automated SAP Fiori Apps Testing with Continuous Delivery

[SAP-samples/teched2020-DEV267](https://github.com/SAP-samples/teched2020-DEV267) - ðŸŒŸ 13

## pactflow/example-consumer-cypress

Demonstrate running a consumer pact test with Cypress

[pactflow/example-consumer-cypress](https://github.com/pactflow/example-consumer-cypress) - ðŸŒŸ 13

## sapmentors/cds-scp-api

CDS Extension for SAP Cloud Platform API Consumption

[sapmentors/cds-scp-api](https://github.com/sapmentors/cds-scp-api) - ðŸŒŸ 12

## rbutera/greater-gatsby

BAREBONES and LIGHTWEIGHT GatsbyJS Default Starter with Typescript, PostCSS, Storybook, & Tailwind CSS

[rbutera/greater-gatsby](https://github.com/rbutera/greater-gatsby) - ðŸŒŸ 12

## freshollie/fresh-configurator

A Work In Progress Betaflight Configurator rewrite. A tool designed to configure flight controllers.

[freshollie/fresh-configurator](https://github.com/freshollie/fresh-configurator) - ðŸŒŸ 12

## sapmentors/sitregcapm

SITreg implemented in the SAP Cloud Application Programming Model

[sapmentors/sitregcapm](https://github.com/sapmentors/sitregcapm) - ðŸŒŸ 11

## linkedpipes/applications

ðŸ–¥ LinkedData based Applications generator

[linkedpipes/applications](https://github.com/linkedpipes/applications) - ðŸŒŸ 11

## OpenCLTBrigade/cmpd-holiday-gift

CMPD Explorers Christmas Project

[OpenCLTBrigade/cmpd-holiday-gift](https://github.com/OpenCLTBrigade/cmpd-holiday-gift) - ðŸŒŸ 11

## jhudsl/fitbitr

Make interacting with the fitbit api from R/shiny sweet, not bitr. 

[jhudsl/fitbitr](https://github.com/jhudsl/fitbitr) - ðŸŒŸ 11

## ianagbip1oti/mat-che

Anonymous Chat. No History.

[ianagbip1oti/mat-che](https://github.com/ianagbip1oti/mat-che) - ðŸŒŸ 11

## angeloocana/ptz-i18n

lib for i18n functions, used in gatsby projects with gatsby-plugin-i18n

[angeloocana/ptz-i18n](https://github.com/angeloocana/ptz-i18n) - ðŸŒŸ 10"
`;
